### PR TITLE
dlna: also look at "Subs" subdirectory, don't swallow some subs, fix mime types

### PR DESCRIPTION
**Pull request by flokli**
Apparently it seems pretty common for subtitles to be put in a subdirectory called "Subs", rather than in the same directory as the media file itself.

This covers that usecase, by checking the returned listing for a directory called "Subs" to exist.

If it does, its child nodes are added to the list before they're being passed to mediaWithResources, allowing these subtitles to be discovered automatically.

#### What is the purpose of this change?

Automatically find subtitles that live in a "Subs" directory alongside the video file, which seems fairly common.

#### Was the change discussed in an issue or in the forum before?

Unknown

#### Checklist

- [x] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-new-feature-or-bug-fix).
- [x] I have added tests for all changes in this PR if appropriate.
- [ ] I have added documentation for the changes if appropriate.
- [x] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [x] I'm done, this Pull Request is ready for review :-)

### CODE REVIEW

**Comment by flokli** [comment]
I added a test for the `mediaWithResources` function to test both the current behaviour, as well as the behaviour with the subtitles inside the `Subs` directory.

**Comment by flokli** [comment]
I also tried this with an example of .idx/.srt files, and noticed the code was swallowing either one of .idx/.sub, and did always send text/srt as a mime type, which is only correct for .srt files. I added these to other individual commits that also extend the test cases. Happy to also move this to a followup PR, just let me know.

**Comment by flokli** [comment]
Hmmh, curious, the test logs mention `application/x-subrip` rather than `text/srt`.

@dwalters (https://github.com/rclone/rclone/commit/6337cc70d320868f47995fdf89290dd845f9b47d), do you recall where `text/srt` comes from?

**Comment by flokli** [comment]
I checked the mime database, and verified `application/x-subrip` again. The samsung TV here is apply with `application/x-subrip` or `text/srt`, so I don't think it matters, and I updated the test cases and the "mime database lite" we ship for some platforms.

**Comment by flokli** [comment]
I dropped checking for specific mime types in the tests, it seems this is still too platform-specific. PTAL.

**Comment by flokli** [comment]
poke @ncw :-) any opinion on that? I've been flying with this patch for a while now, but noticed this PR is still open. Would be nice to see it landing upstream.

File: cmd/serve/dlna/cds.go [hunk 1]
```diff
@@ -130,6 +130,20 @@ func (cds *contentDirectoryService) readContainer(o object, host string) (ret []
 		return
 	}
 
+	// if there's a "Subs" child directory, add its children to the list as well,
+	// so mediaWithResources is able to find them.
+	for _, node := range dirEntries {
+		if node.Name() == "Subs" && node.IsDir() {
```
Comment (ncw) [inline cmd/serve/dlna/cds.go:151]: This should probably be a case insensitive compare so `strings.EqualFold`

Comment (ncw) [inline cmd/serve/dlna/cds.go:151]: Does this directory name need to be configured?

It certainly needs documenting though - a line in the docs about here maybe?

https://github.com/rclone/rclone/blob/70e8ad456fcc627f57c001aa82ac925bd9fb47a8/cmd/serve/dlna/dlna.go#L50

**Review by ncw — COMMENTED** [review]
Sorry this didn't get reviewed before.

These look like nice changes. I put a small query inline but otherwise great - thank you :-)

File: cmd/serve/dlna/cds.go (hunk 1 above)
Comment (flokli) [reply cmd/serve/dlna/cds.go:151]: They didn't mention the subtitle discovery yet at all, but yes, can add.

`Subs` is a convention I've seen so far quite recently, and so far nothing else seems to be commonly used. I don't think it makes a lot of sense to make this configurable.

Comment (flokli) [reply cmd/serve/dlna/cds.go:151]: Ok, will do.

Comment (flokli) [reply cmd/serve/dlna/cds.go]: Added a commit documenting the functionality.

Comment (flokli) [reply cmd/serve/dlna/cds.go]: Done

**Review by ncw — APPROVED** [review]
That looks great - thank you :-)

I'll merge now.

**Pull request merged**